include/nbl/video/SPhysicalDeviceLimits.h

```diff
@@ -0,0 +1,497 @@
+    //uint32_t           simdPerComputeUnit;
+    //uint32_t           wavefrontsPerSimd;
+    //uint32_t           wavefrontSize;
+    //uint32_t           sgprsPerSimd;
+    //uint32_t           minSgprAllocation;
+    //uint32_t           maxSgprAllocation;
+    //uint32_t           sgprAllocationGranularity;
+    //uint32_t           vgprsPerSimd;
+    //uint32_t           minVgprAllocation;
+    //uint32_t           maxVgprAllocation;
+    //uint32_t           vgprAllocationGranularity;
+
+    /* ShaderCoreProperties2AMD *//* provided by VK_AMD_shader_core_properties2 */
+    //VkShaderCorePropertiesFlagsAMD    shaderCoreFeatures;
+    //uint32_t                          activeComputeUnitCount;
+
+    /* BlendOperationAdvancedPropertiesEXT *//* provided by VK_EXT_blend_operation_advanced */
+    //uint32_t           advancedBlendMaxColorAttachments;
+    //VkBool32           advancedBlendIndependentBlend;
+    //VkBool32           advancedBlendNonPremultipliedSrcColor;
+    //VkBool32           advancedBlendNonPremultipliedDstColor;
+    //VkBool32           advancedBlendCorrelatedOverlap;
+    //VkBool32           advancedBlendAllOperations;
+            
+    /* ConservativeRasterizationPropertiesEXT *//* provided by VK_EXT_conservative_rasterization */
+    //float              primitiveOverestimationSize;
+    //float              maxExtraPrimitiveOverestimationSize;
+    //float              extraPrimitiveOverestimationSizeGranularity;
+    //VkBool32           primitiveUnderestimation;
+    //VkBool32           conservativePointAndLineRasterization;
+    //VkBool32           degenerateTrianglesRasterized;
+    //VkBool32           degenerateLinesRasterized;
+    //VkBool32           fullyCoveredFragmentShaderInputVariable;
+    //VkBool32           conservativeRasterizationPostDepthCoverage;
+            
+    /* CustomBorderColorPropertiesEXT *//* provided by VK_EXT_custom_border_color */
+    //uint32_t           maxCustomBorderColorSamplers;
+
+    /* DescriptorIndexingPropertiesEXT ---> MOVED TO Vulkan 1.2 Core  *//* provided by VK_AMD_shader_core_properties */
+
+    /* DiscardRectanglePropertiesEXT *//* provided by VK_EXT_discard_rectangles */
+    //uint32_t           maxDiscardRectangles;
+            
+    /* ExternalMemoryHostPropertiesEXT *//* provided by VK_EXT_external_memory_host */
+    //VkDeviceSize       minImportedHostPointerAlignment;
+    
+    /* FragmentDensityMapPropertiesEXT *//* provided by VK_EXT_fragment_density_map */
+    //VkExtent2D         minFragmentDensityTexelSize;
+    //VkExtent2D         maxFragmentDensityTexelSize;
+    //VkBool32           fragmentDensityInvocations;
+    
+    /* FragmentDensityMap2PropertiesEXT *//* provided by VK_EXT_fragment_density_map2 */
+    //VkBool32           subsampledLoads;
+    //VkBool32           subsampledCoarseReconstructionEarlyAccess;
+    //uint32_t           maxSubsampledArrayLayers;
+    //uint32_t           maxDescriptorSetSubsampledSamplers;
```

Comment:
I have no plan to expose these on GL & GLES, so treat them like a Vulkan device that doesn't support these extensions

Reply:
But you previously said:
> not implementing or exposing VRS in near or far future
So should I expose and set to false or just keep commented? I'm confused

Reply:
I'll leave as TODO
What does this `VK_EXT_fragment_density_map` has to do with VRS?

Reply:
> I'll leave as TODO What does this `VK_EXT_fragment_density_map` has to do with VRS?

>> This extension allows an application to specify areas of the render target where the fragment shader may be invoked fewer times. These fragments are broadcasted out to multiple pixels to cover the render target.

the definition of VRS

Reply:
So does these two extensions count as `not implementing or exposing VRS in near or far future`?

Reply:
> So does these two extensions count as `not implementing or exposing VRS in near or far future`?

yes